logemongo

```diff
@@ -96,20 +102,40 @@ while ($OPTIONS{count} ? $counter <= $OPTIONS{count} : 1) {
 		}
 		my ($len, $msg) = split "\r\n", $buf;
 
+		my ($logdata, $payload, %msghash);
+		if ($msg =~ m/(\[NXLOG@\d+(.*)\]) (.*)/) {
+			$logdata = $2;
+			$payload = $3;
+		}
+		my @time_uuid = split ' ', $msg;
+		($msghash{'timestamp'}, $msghash{'uuid'}) =  ($time_uuid[0], $time_uuid[1]);
+		my $qr = qr/\s*([^=,\s]+)\s*=\s*(?:"((?:[^"]|"")*)"|([^,"]*))\s*,?/;
+		while ($logdata =~ m/$qr/) {
+			$msghash{$1}=$2;
+			$logdata =~ s/$qr//;
+		}
+		$msghash{'payload'} = $payload;
+
 		if (@INCLUDES) {
 			for my $re (@INCLUDES) {
-				goto g2g if $msg =~ $re;
+				goto g2g if $payload =~ $re;
 			}
 			next message;
 		}
 		if (@EXCLUDES) {
 			for my $re (@EXCLUDES) {
-				goto g2g if $msg !~ $re;
+				goto g2g if $payload !~ $re;
+			}
+			next message;
```

Comment:
I think the disposition on this loop is wrong.  It seems to be saying that the message is good if the first exclusion doesn't match, not that the message is good if _NONE_ of the exclusions match.